OpenCabProvider/app/src/main/java/com/eleostech/exampleprovider/MainActivity.java

```diff
@@ -28,142 +32,195 @@
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+
+        binding.identityProviderTokenTypeSwitch.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                updateIdentityProviderTokenType();
+            }
+        });
+
+        binding.identityProviderTokenTextedit.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                saveToken(s.toString());
+            }
+        });
+    }
+
+    private void saveToken(String text) {
+        Preferences.setIdentityResponseToken(this, text);
+    }
+
+    private void updateIdentityProviderTokenType() {
+        Preferences.setIdentityResponseJWT(this, binding.identityProviderTokenTypeSwitch.isChecked());
+        if (binding.identityProviderTokenTypeSwitch.isChecked()) {
+            binding.identityProviderTokenTextedit.setVisibility(View.GONE);
+            Preferences.setIdentityResponseToken(this, "");
+        } else {
+            binding.identityProviderTokenTextedit.setVisibility(View.VISIBLE);
+        }
     }
 
+
     @Override
     public void onResume() {
         super.onResume();
 
         boolean isNav = Preferences.getNavigationState(this);
-        if(isNav) {
-            navigationState.setText("NAVIGATING: TRUE");
+        if (isNav) {
+            binding.navigationState.setText("NAVIGATING: TRUE");
         } else {
-            navigationState.setText("NAVIGATING: FALSE");
+            binding.navigationState.setText("NAVIGATING: FALSE");
         }
     }
 
     private void login(final String username) {
         Log.d(LOG_TAG, "login()");
 
         Preferences.setUsername(MainActivity.this, username);
-        welcomeText.setText("Welcome " + username);
-        loginContainer.setVisibility(View.GONE);
-        logoutContainer.setVisibility(View.VISIBLE);
+        binding.welcomeUser.setText("Welcome " + username);
+        binding.loginContainer.setVisibility(View.GONE);
+        binding.logoutContainer.setVisibility(View.VISIBLE);
         setDutyStatus("off");
-        addUser(username, true);
-
+        addUser(username, false);
+
+        List<PackageInfo> packages = getPackageManager().getInstalledPackages(PackageManager.GET_RECEIVERS);
+        search:
+        for (PackageInfo pkg : packages) {
+            if (pkg.receivers != null) {
+                for (ActivityInfo activityInfo : pkg.receivers) {
+                    if (activityInfo.name.endsWith(".IdentityChangedReceiver")) {
```

Comment:
Here's the other key change. To allow consumer apps to be woken up by the OS to react to new information even if they are not running, broadcast intents are now _explicit_ across all registered receiver apps, rather than being implicit and only usable by actively-running background apps.